scaffold-eth/create-eth
# Configure eslint and prettiier

### Description : 

- Configured prettier inline with SE-2 repo config
- Formatted files with prettier
- Configured [ESlint v9](https://eslint.org/docs/latest/use/getting-started) and [typescript-eslint](https://typescript-eslint.io/getting-started/)
- Both prettier and eslint ignores templates dir
- Fixed eslint issues
- Configured pre-commit hooks with husky and lint-staged


Tried keeping typescript-eslint rules inline with SE-2. 

## To test : 
1. Install dependencies : 
```
yarn install
```

2. Add `const x = 5` in `src/cli.ts` and try running : 
```
git add . && git commit -m "this should fail"
```

This will run pre-commit and hopefully it should fail (since x is unused variable) 

cc @Pabl0cks, could you please test this on windows also once, because I remmember in SE-2 while configuring husky it was not working properly especially for windows. 

## New Commands : 

To run lint : 

```
yarn lint
```

or to fix all fixable errors:  `yarn lint --fix`

To format files with prettier : 

```
yarn format
```
